pkg/client/api_client.go

```diff
@@ -147,13 +157,14 @@ func (c *APIClient) findTarget(target *api.Target) (*api.Target, error) {
 
 	// Iterate over the list of targets to look for the given target
 	// by comparing the category, target type and identifier fields
+	// target type is regarded the same if the old one only differs by an extra suffix
 	for _, tgt := range targetList {
 		// array of InputFields
 		for _, inputField := range tgt.InputFields {
 			if inputField.Name == "targetIdentifier" &&
 				inputField.Value == targetId &&
 				tgt.Category == target.Category &&
-				tgt.Type == target.Type {
+				strings.HasPrefix(tgt.Type, target.Type) {
```

Comment:
I am a bit slow on this, but why does this logic only apply to `api_client` but not `tp_client`?

Reply:
That's because the `findTarget()` method in `tp_client` doesn't compare the probe type and just relies on the `targetName` to fetch.